src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -1289,14 +1289,14 @@ Symbol getAttributeTarget(SyntaxNode targetSyntax)
             {
                 return targetSyntax switch
                 {
-                    BaseMethodDeclarationSyntax methodDeclaration => GetDeclaredMemberSymbol(methodDeclaration),
-                    LocalFunctionStatementSyntax localFunction => GetMemberModel(localFunction)?.GetDeclaredLocalFunction(localFunction),
-                    ParameterSyntax parameterSyntax => ((Symbols.PublicModel.ParameterSymbol)GetDeclaredSymbol(parameterSyntax)).UnderlyingSymbol,
-                    TypeParameterSyntax typeParameterSyntax => ((Symbols.PublicModel.TypeParameterSymbol)GetDeclaredSymbol(typeParameterSyntax)).UnderlyingSymbol,
-                    IndexerDeclarationSyntax indexerSyntax => ((Symbols.PublicModel.PropertySymbol)GetDeclaredSymbol(indexerSyntax)).UnderlyingSymbol,
-                    AccessorDeclarationSyntax accessorSyntax => ((Symbols.PublicModel.MethodSymbol)GetDeclaredSymbol(accessorSyntax)).UnderlyingSymbol,
-                    AnonymousFunctionExpressionSyntax anonymousFunction => ((Symbols.PublicModel.Symbol)GetSymbolInfo(anonymousFunction).Symbol).UnderlyingSymbol,
-                    DelegateDeclarationSyntax delegateSyntax => ((Symbols.PublicModel.NamedTypeSymbol)GetDeclaredSymbol(delegateSyntax)).UnderlyingSymbol,
+                    BaseMethodDeclarationSyntax or
+                        LocalFunctionStatementSyntax or
+                        ParameterSyntax or
+                        TypeParameterSyntax or
+                        IndexerDeclarationSyntax or
+                        AccessorDeclarationSyntax or
+                        DelegateDeclarationSyntax => GetDeclaredSymbolForNode(targetSyntax).GetSymbol(),
+                    AnonymousFunctionExpressionSyntax anonymousFunction => GetSymbolInfo(anonymousFunction).Symbol.GetSymbol(),
                     _ => null
```

Comment:
Any harm calling GetDeclaredSymbolForNode for whatever syntax we have?

```suggestion
                    AnonymousFunctionExpressionSyntax anonymousFunction => GetSymbolInfo(anonymousFunction).Symbol.GetSymbol(),
                    _ => GetDeclaredSymbolForNode(targetSyntax).GetSymbol()
```

Reply:
I'd rather enumerate the valid cases